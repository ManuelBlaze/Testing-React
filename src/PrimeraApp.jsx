// Functional Components
import React from 'react';
import PropTypes from "prop-types";
// import React, { Fragment } from 'react';

const PrimeraApp = ( { saludo, subtitulo } ) => {

    // const saludo = {
    //     nombre: 'Manuel',
    //     edad: 21
    // };
    


    return (
        <>
            <h1> { saludo } </h1>
            {/* <pre> { JSON.stringify( saludo, null, 3 ) } </pre> */}
            <p> { subtitulo } </p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Mi primera App'
}


export default PrimeraApp;