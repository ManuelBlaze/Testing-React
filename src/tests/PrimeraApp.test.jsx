import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";

import PrimeraApp from "../PrimeraApp";

describe('Tests in <PrimeraApp />', () => {
    
    // test('should show "Hola Mundo"', () => {    
    //     const title = 'Hola Mundo';
    //     const { getByText } = render( <PrimeraApp saludo = 'Hola Mundo'/> );
    //     expect( getByText(title) ).toBeInTheDocument();
    // });
    
    test('should show <PrimeraApp /> correctly', () => {
        
        const mes = 'Hola Mundo'
        const wrapper = shallow( <PrimeraApp saludo = { mes } /> );
        
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should show the subtitle send it by props', () => {
        
        const mes = 'Hola Mundo';
        const sub = ' Soy un subtitulo ';

        const wrapper = shallow( 
            <PrimeraApp 
                saludo = { mes }
                subtitulo = { sub } 
            /> 
        );

        const text = wrapper.find('p').text();
        
        expect( text ).toBe( sub );
    });
    
});