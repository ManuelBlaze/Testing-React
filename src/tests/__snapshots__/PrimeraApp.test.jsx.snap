// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <PrimeraApp /> should show <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h1>
     
    Hola Mundo
     
  </h1>
  <p>
     
    Mi primera App
     
  </p>
</Fragment>
`;
